plugins/Rockets/jsonPropertyMap.h

```diff
@@ -552,25 +753,128 @@ inline bool from_json(brayns::PropertyMap& obj, const std::string& json)
+        case PropertyMap::Property::Type::Double:
+        {
+            double val;
+            if (!get_property(m.value, val))
+                return false;
+            obj.updateProperty(propName, val);
+            break;
+        }
         case PropertyMap::Property::Type::Int:
-            SET_PROPERTY(Int)
+            if (!parseSetProperty<int32_t>(m.value, obj, propName))
+                return false;
+            break;
         case PropertyMap::Property::Type::String:
-            SET_PROPERTY(String)
+            if (!parseSetProperty<std::string>(m.value, obj, propName))
+                return false;
+            break;
         case PropertyMap::Property::Type::Bool:
-            SET_PROPERTY(Bool)
-        case PropertyMap::Property::Type::Vec2f:
-            SET_ARRAY(double, Double, 2)
+            if (!parseSetProperty<bool>(m.value, obj, propName))
+                return false;
+            break;
+        case PropertyMap::Property::Type::Vec2d:
+            if (!parseSetArray<double, 2>(m.value, obj, propName))
+                return false;
+            break;
         case PropertyMap::Property::Type::Vec2i:
-            SET_ARRAY(int32_t, Int, 2)
-        case PropertyMap::Property::Type::Vec3f:
-            SET_ARRAY(double, Double, 3)
+            if (!parseSetArray<int32_t, 2>(m.value, obj, propName))
+                return false;
+            break;
+        case PropertyMap::Property::Type::Vec3d:
+            if (!parseSetArray<double, 3>(m.value, obj, propName))
+                return false;
+            break;
         case PropertyMap::Property::Type::Vec3i:
-            SET_ARRAY(int32_t, Int, 3)
-        case PropertyMap::Property::Type::Vec4f:
-            SET_ARRAY(double, Double, 4)
+            if (!parseSetArray<int32_t, 3>(m.value, obj, propName))
+                return false;
+            break;
+        case PropertyMap::Property::Type::Vec4d:
+            if (!parseSetArray<double, 4>(m.value, obj, propName))
+                return false;
+            break;
         }
     }
     return true;
 }
+
+template <>
+inline std::string to_json(const brayns::ModelProperties& props)
+{
+    const auto replaceFirstOccurrence = [](std::string s,
+                                           const std::string& toReplace,
+                                           const std::string& replaceWith) {
+        std::size_t pos = s.find(toReplace);
+        if (pos == std::string::npos)
+            return s;
+        return s.replace(pos, toReplace.length(), replaceWith);
+    };
+
+    const auto jsonOriginal = staticjson::to_json_string(props);
+
+    const std::string propertiesJson =
+        "\"properties\":" + to_json(props.properties);
+
+    const auto result =
+        replaceFirstOccurrence(jsonOriginal, "\"properties\":{}",
+                               propertiesJson);
+
+    return result;
+}
+
+brayns::PropertyMap jsonToPropertyMap(const std::string& json)
```

Comment:
what's the difference to `bool from_json(brayns::PropertyMap& obj, const std::string& json)`?

Reply:
The difference is that ```jsonToPropertyMap``` creates a property map from scratch filling in everything from the json and infers the type of the properties. ```from_json``` on the other hand fills in an already existing property map with the types already known.